worker/src/RTC/RtpStreamRecv.cpp

```diff
@@ -72,6 +72,70 @@ namespace RTC
 		return true;
 	}
 
+	bool RtpStreamRecv::ReceiveRtxPacket(RTC::RtpPacket* packet)
+	{
+		MS_TRACE();
+
+		MS_ASSERT(packet->GetSsrc() == this->rtxSsrc, "invalid ssrc on rtx packet");
+
+		// RTP packet sanity check.
+		if (packet->GetPayloadLength() < 2)
+		{
+			MS_WARN_TAG(rtx, "ignoring rtx packet with no space for a rtx header field [ssrc: %" PRIu32 " seqnr: %" PRIu16 " payload type: %" PRIu8 "]",
+					packet->GetSsrc(),
+					packet->GetSequenceNumber(),
+					packet->GetPayloadType());
+
+			return false;
+		}
+
+		// Check that the payload type corresponds to the one negotiated.
+		if (packet->GetPayloadType() != this->rtxPayloadType)
+		{
+			MS_WARN_TAG(rtx, "ignoring rtx packet with invalid payload type [ssrc: %" PRIu32 " seqnr: %" PRIu16 " payload type: %" PRIu8 "]",
+					packet->GetSsrc(),
+					packet->GetSequenceNumber(),
+					packet->GetPayloadType());
+
+			return false;
+		}
+
+		// Get the rtx packet sequence number for logging purposes.
+		this->rtxSeq = packet->GetSequenceNumber();
+
+		// Get the original rtp packet.
+		packet->RtxDecode(this->params.payloadType, this->params.ssrc);
+
+		MS_DEBUG_TAG(rtx, "received rtx packet [ssrc: %" PRIu32 " seqnr: %" PRIu16 "] recovering original [ssrc: %" PRIu32 " seqnr: %" PRIu16 "]",
+				this->rtxSsrc,
+				this->rtxSeq,
+				packet->GetSsrc(),
+				packet->GetSequenceNumber());
+
+		// Set the extended sequence number into the packet.
+		packet->SetExtendedSequenceNumber(
+		    this->cycles + static_cast<uint32_t>(packet->GetSequenceNumber()));
+
+		// Set RTP header extension ids.
```

Comment:
This is duplicated; used in ReceivePacket as well. Create a private SetHeaderExtensions method.